capstone_project/src/main/java/com/google/sps/data/PostService.java

```diff
@@ -200,4 +196,58 @@ private float getUpvoteRatio(Entity post) {
 
     return upvotes / minDifference;
   }
+
+  /**
+   * Increases the reaction count of the submitted reaction by one. Returns the new reaction count
+   * after increase.
+   */
+  public long reactToPost(HttpServletRequest request) {
+    String reaction = request.getParameter("reaction");
+    if (reaction == null || reaction.isEmpty()) {
+      return -1;
```

Comment:
Are these valid reaction counts?

If these -1 are used as a "special value", I would avoid doing that. If you want the return type to signify that the post doesn't have score because it doesn't exist, you can either.

* Throw and exception, probably create your own exception that says the post/reaction does not exist
* OR change the return type to Optional<Long> and in the method comment say that if Optional.empty() is returned, then the post/reaction does not exist.

Reply:
I modified the methods to use Optionals instead of returning a negative value